.circleci/config.yml

```diff
@@ -18,13 +13,102 @@ publish_with_tags_filter: &publish_with_tags_filter
     tags:
       only: /^v[0-9]+\.[0-9]+\.[0-9]+/
 
+parameters:
+  # used to trigger the workflow
+  trigger:
+    type: boolean
+    default: true
+
+  # parameter per package
+  ef-runtime-client:
+    type: boolean
+    default: false
+
+  ef-demo-component:
+    type: boolean
+    default: false
+
+executors:
+  node:
+    docker:
+      - image: cimg/node:18.16
+
 jobs:
-  publish:
-    <<: *defaults
+  trigger-workflows:
+    executor: node
+    steps:
+      - checkout
+      - run:
+          name: Trigger workflows
+          command: chmod +x .circleci/circle_trigger.sh && .circleci/circle_trigger.sh
+
+  build_test_package:
+    parameters:
+      package_name:
+        type: string
+
+    executor: node
+    working_directory: ~/project/packages/<< parameters.package_name >>
+
+    steps:
+      - checkout:
+          path: ~/project
+      - run:
+          name: Install
+          command: |
+            npm install
+      - run:
+          name: Test
+          command: |
+            npm test
+      - run:
+          name: Build
+          command: |
+            npm run build
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - packages/<< parameters.package_name >>
+
+  build_test_example:
+    parameters:
+      example_name:
+        type: string
+
+    executor: node
+    working_directory: ~/project/examples/<< parameters.example_name >>
+
+    steps:
+      - checkout:
+          path: ~/project
+      - run:
+          name: Install
+          command: |
+            npm install
+      - run:
+          name: Test
+          command: |
+            npm test
+      - run:
+          name: Build
+          command: |
+            npm run build
+      - persist_to_workspace:
+          root: ~/project
+          paths:
+            - examples/<< parameters.example_name >>
+
+  publish_package:
+    parameters:
+      package_name:
+        type: string
```

Comment:
is this param used to confirm if we are pushing a new version to npm?

Reply:
No, This is a job parameter that is passed in the workflows later per package. Assuming that we will have multiple packages in the monorepo.

Reply:
No, this is a parameter to run the generic job. The parameter is set in the workflow per package later.